include/dali/core/mm/async_pool.h

```diff
@@ -215,7 +215,10 @@ class async_pool_resource : public async_memory_resource<Kind> {
       min_align = 256;
     if (min_align > alignment)
       alignment = min_align;
-    size = align_up(size, alignment);
+    size_t aligned = align_up(size, alignment);
+    if (check && aligned < size)
```

Comment:
Why you want to throw here? I would assume that alignment should be smaller than size, not the oposite?

Reply:
NVM. I see now,

Reply:
It's not `alignment` - it's `aligned` - this means that (most likely) after aligning the aligned overflowed and is now 0.